tests/clients/test_pep0249.py

```diff
@@ -0,0 +1,320 @@
+            for record in sample_seq_of_params:
+                assert record in response
+            assert len(response) == 3
+
+    def test_executemany_with_tracing(self):
+        with tracer.start_as_current_span("test"):
+            sample_sql = """insert into tests (id, name, email) values (%s, %s, %s) returning id, name, email;"""
+            sample_seq_of_params = [
+                (6, "sample-name-3", "sample-email-3@mail.com"),
+                (7, "sample-name-4", "sample-email-4@mail.com"),
+            ]
+            self.test_wrapper.executemany(sample_sql, sample_seq_of_params)
+
+            # Exception Handling
+            with pytest.raises(Exception) as exc_info, patch.object(
+                CursorWrapper, "_collect_kvs", side_effect=Exception("test exception")
+            ) as mock_collect_kvs:
+                self.test_wrapper.executemany(
+                    sample_sql, seq_of_parameters=sample_seq_of_params
+                )
+            assert str(exc_info.value) == "test exception"
+            mock_collect_kvs.assert_called_once()
+            self.test_wrapper.execute("select * from tests;")
+            response = self.test_wrapper.fetchall()
+            for record in sample_seq_of_params:
+                assert record in response
+            assert len(response) == 3
+
+    def test_callproc_with_tracing_off(self):
+        with tracer.start_as_current_span("sqlalchemy"):
+            sample_proc_name = "call insert_user(%s, %s, %s);"
+            sample_params = (8, "sample-name-8", "sample-email-8@mail.com")
+            self.test_wrapper.callproc(sample_proc_name, sample_params)
+            self.test_conn.commit()
+            self.test_wrapper.execute("select * from tests;")
+            response = self.test_wrapper.fetchall()
+            assert sample_params in response
+            assert len(response) == 2
+
+    def test_callproc_with_tracing(self):
+        with tracer.start_as_current_span("test"):
+            sample_proc_name = "call insert_user(%s, %s, %s);"
+            sample_params = (9, "sample-name-9", "sample-email-9@mail.com")
+            self.test_wrapper.callproc(sample_proc_name, sample_params)
+            self.test_conn.commit()
+            self.test_wrapper.execute("select * from tests;")
+            response = self.test_wrapper.fetchall()
+            assert sample_params in response
+            assert len(response) == 2
+
+            # Exception Handling
+            error_proc_name = "erroroeus command;"
+            with pytest.raises(Exception) as exc_info, patch.object(
+                InstanaSpan,
+                "record_exception",
+            ) as mock_exception:
+                self.test_wrapper.callproc(error_proc_name, sample_params)
+            assert exc_info.typename == "SyntaxError"
+            mock_exception.call_count == 2
+
+
+class TestConnectionWrapper:
+    @pytest.fixture(autouse=True)
+    def _setup(self) -> Generator[None, None, None]:
```

Comment:
Same as [this](https://github.com/instana/python-sensor/pull/575/files#r1713859847) comment.

Reply:
Done.